monitoring/uss_qualifier/scd/simulator/operational_intents_flight_authorisation_data_generator.py

```diff
@@ -0,0 +1,373 @@
+        
+        flight_path_geometry = self._generate_single_flight_geometry(geometry_generation_rule = geometry_generation_rule, injection_number= injection_number)
+
+        raw_geometry = GeneratedGeometry(geometry = flight_path_geometry, geometry_generation_rule = geometry_generation_rule)
+        
+        return raw_geometry
+
+
+    def generate_astm_4d_volumes(self,raw_geometry:GeneratedGeometry, altitude_of_ground_level_wgs_84 :int) -> Volume4D:
+        """A method to generate ASTM specified Volume 4D payloads to submit to the system to be tested.  """
+        
+        flight_volume_3d = self.convert_geometry_to_volume(flight_geometry = raw_geometry.geometry, altitude_of_ground_level_wgs_84 = altitude_of_ground_level_wgs_84)
+        flight_volume_4d = self.transform_3d_volume_to_astm_4d(volume_3d = flight_volume_3d)
+        
+        return flight_volume_4d
+    
+def generate_injection_operational_intent(astm_4d_volume:Volume4D) -> OperationalIntentTestInjection:
+    """A method to generate Operational Intent injection  """        
+    current_operational_intent_reference = OperationalIntentTestInjection(volumes = [astm_4d_volume], state = 'Accepted', off_nominal_volumes = [], priority =0)            
+    return current_operational_intent_reference
+
+class KnownIssuesAcceptableResultFieldGenerator():
+    """A class to generate Known Issues and Acceptable results as a result of data processing of the test data provided by the USSP"""
+    
+    def __init__(self, expected_flight_authorisation_processing_result:str, expected_operational_intent_processing_result:str):
+
+        self.expected_flight_authorisation_processing_result = expected_flight_authorisation_processing_result
+        self.expected_operational_intent_processing_result = expected_operational_intent_processing_result        
+        self.all_test_results = {expected_flight_authorisation_processing_result, expected_operational_intent_processing_result}
+        self.acceptable_results = list(self.all_test_results)
+
+        if "Planned" in self.all_test_results and len(self.all_test_results) != 1:
+            self.acceptable_results.remove("Planned")
+
+
+    def generate_known_issue_fields(self, incorrect_field:str = None)-> Dict[str, KnownIssueFields]: 
+        """A method to generate messages for the user to take remedial actions """
+
+        all_known_issue_fields = {}
+
+        common_error_notification = KnownIssueFields(test_code = "flight_authorisation_test", 
+                                                     relevant_requirements = ["A correct UAS Serial number is equired by ANNEX IV of COMMISSION IMPLEMENTING REGULATION (EU) 2021/664, paragraph 1", "Registration number of the UAS operator. The format is defined in EASA Easy Access Rules for Unmanned Aircraft Systems GM1 to AMC1 Article 14(6) Registration of UAS operators and ‘certified’ UAS. Required by ANNEX IV of COMMISSION IMPLEMENTING REGULATION (EU) 2021/664, paragraph 10."], 
```

Comment:
typo: "equired"

Reply:
Thank you, the `relevant_requirements` text has been removed per the earlier comment.